aiutils/vis/image_io.py

```diff
@@ -7,27 +7,22 @@ def imread(filename):
     Matlab like function for reading an image file.
 
     Returns:
-    im_array (numpy.ndarray): h x w x 3 ndarray for color images and 
-        h x w for grayscale images
+      im_array (numpy.ndarray): See the following link for information on
+      different modes and the array sizes returned in those modes:
+      http://pillow.readthedocs.io/en/3.4.x/handbook/concepts.html#modes
     """
     im = Image.open(filename)
-
-    err_str = \
-        "imread only supports 'RGB' and 'L' modes, found '{}'".format(im.mode)
-    assert (im.mode == 'RGB' or im.mode == 'L'), err_str
-
     im_array = np.array(im)
-
-    return im_array
+    return im_array, im
```

Comment:
So say the read image was a weird PIL mode like a `P` (as is the case with `.gif`) or `RGBA`. Then while saving or displaying we need to know the mode. If we have the original PIL object `im`, then extracting mode is simply `im.mode`.  So why not simply return `im.mode`? Because `im.show()` and `im.save()` are convenient functions as well. PIL also supports other image operations that need this object. 
What do you think?